### PR TITLE
fix: remove download button when content is empty

### PR DESCRIPTION
**Pull request by kowalczyk-krzysztof**
This PR fixes an edge case where if user specifies `content-type` as `application/octet-stream` and `content-length` as `0` the `Download file` button appears even though there is nothing to download.

### CODE REVIEW

**Review by tekyu — APPROVED** [review]

File: src/core/components/response-body.jsx [hunk 1]
```diff
@@ -59,6 +59,7 @@ export default class ResponseBody extends React.PureComponent {
     url = url || ""
 
     if (
+      content.size > 0 &&
```
Comment (char0n) [inline src/core/components/response-body.jsx:62]: When this `if` statement executes and `content.size > 0` expression is the first predicate, it is not guaranteed that the `content` is of type `Blob`. In order to guarantee that the `content` is of type `Blobl` we have to change the order of the predicates in the following way:

```js
if (
  (/^application\/octet-stream/i.test(contentType) ||
    (headers["Content-Disposition"] &&
      /attachment/i.test(headers["Content-Disposition"])) ||
    (headers["content-disposition"] &&
      /attachment/i.test(headers["content-disposition"])) ||
    (headers["Content-Description"] &&
      /File Transfer/i.test(headers["Content-Description"])) ||
    (headers["content-description"] &&
      /File Transfer/i.test(headers["content-description"]))) &&
  content.size > 0
) {
```

Comment (char0n) [reply src/core/components/response-body.jsx:62]: Let's put a couple of tests in at the end of **test/unit/components/response-body.jsx**:

```jsx
  it("should render Download file link for non-empty response", function () {
    props.contentType = "application/octet-stream"
    props.content = new Blob(['"test"'], { type: props.contentType });
    const wrapper = shallow(<ResponseBody {...props} />)
    expect(wrapper.text()).toMatch(/Download file/)
  })

  it("should not render Download file link for empty response", function () {
    props.contentType = "application/octet-stream"
    props.content = new Blob()
    const wrapper = shallow(<ResponseBody {...props} />)
    expect(wrapper.text()).not.toMatch(/Download file/)
  })
```

Comment (char0n) [reply src/core/components/response-body.jsx:62]: In order for these tests to work you need to modify `setUpDomEnvironment` function in **test/unit/setup.js** and add support for creating URL objects:

```js
  // https://github.com/jsdom/jsdom/issues/1721
  if (typeof global.window.URL.createObjectURL === 'undefined') {
    Object.defineProperty(global.window.URL, 'createObjectURL', { value: () => 'data:text/plain;base64,SGVsbG8sIFdvcmxkIQ==' })
  }
```

**Review by char0n — CHANGES_REQUESTED** [review]
Hi @kowalczyk-krzysztof,

Thanks for the PR and bringing up this issue! Please look at the code review suggestions and comments which we need to address before this change gets merged.

File: src/core/components/response-body.jsx (hunk 1 above)
Comment (kowalczyk-krzysztof) [reply src/core/components/response-body.jsx]: Thanks @char0n ! 
Yep, the predicate order makes sense. I ran into some issues with unit tests so I didn't make any but the change in `setUpDomEnvironment` helps. Thanks!

I've pushed the requested changes.

File: test/unit/setup.js [hunk 2]
```diff
@@ -22,6 +22,11 @@ function setUpDomEnvironment() {
   })
   const { window } = jsdom
 
+  // https://github.com/jsdom/jsdom/issues/1721
```
Comment (char0n) [inline test/unit/setup.js:25]: This needs to come after `global.window` is redefined on line `30`. I'll do the change.

**Review by char0n — APPROVED** [review]
LGTM

File: src/core/components/response-body.jsx (hunk 1 above)
Comment (char0n) [reply src/core/components/response-body.jsx]: @kowalczyk-krzysztof thanks for the quick changes.

**Comment by char0n** [comment]
@nick-coremont can you corroborate please?

**Comment by nick-coremont** [comment]
@char0n What do you mean please? I believe the issue I referenced includes the required evidence.

**Comment by char0n** [comment]
@nick-coremont sorry, I've overlooked the link to #9298. Handling it now. Thanks!

**Pull request merged**